### PR TITLE
[Storage] Support excludePrefix, includeDeleted, and new schema fields in blob inventory

### PR DESCRIPTION
**Pull request by yifanz0**
## Description
Add support for ExcludePrefix, IncludeDeleted, and many new schema fields in blob inventory. 
Also reformatted the management plane unit test ps1 file. 

Design doc: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1173

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md [hunk 1]
```diff
@@ -33,11 +34,14 @@ The **New-AzStorageBlobInventoryPolicyRule** cmdlet creates a blob inventory pol
...
 <!-- Skip: Output cannot be splitted from code -->
+
 ```
 PS C:\> $rule1 = New-AzStorageBlobInventoryPolicyRule -Name Test1 -Destination $containerName -Disabled -Format Csv -Schedule Daily -ContainerSchemaField Name,Metadata,PublicAccess,Last-mOdified,LeaseStatus,LeaseState,LeaseDuration,HasImmutabilityPolicy,HasLegalHold -PrefixMatch con1,con2
 
 PS C:\> $rule2 = New-AzStorageBlobInventoryPolicyRule -Name Test2 -Destination $containerName -Format Parquet -Schedule Weekly -IncludeBlobVersion -IncludeSnapshot -BlobType blockBlob,appendBlob -PrefixMatch aaa,bbb `
                 -BlobSchemaField name,Creation-Time,Last-Modified,Content-Length,Content-MD5,BlobType,AccessTier,AccessTierChangeTime,Expiry-Time,hdi_isfolder,Owner,Group,Permissions,Acl,Metadata
+PS C:\> $rule3 = New-AzStorageBlobInventoryPolicyRule -Name Test3 -Destination $containerName -Format Parquet -Schedule Weekly -IncludeBlobVersion -IncludeSnapshot -IncludeDeleted -BlobType blockBlob,appendBlob -PrefixMatch aaa,bbb `
+                 -ExcludePrefix ccc,ddd -BlobSchemaField name, Last-Modified, BlobType,AccessTier,AccessTierChangeTime,Content-Type,Content-CRC64,CopyId,x-ms-blob-sequence-number,TagCount
```
Comment (blueww) [inline src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md:44]: for `name, Last-Modified, BlobType,AccessTier`, there should be no space between 2 schemas, or this won't be take as the value of a single parameter.

File: src/Storage/Storage.Management/StorageAccount/NewAzureStorageBlobInventoryPolicyRule.cs [hunk 2]
```diff
@@ -87,15 +92,47 @@ public class NewAzureStorageBlobInventoryPolicyRuleCommand : StorageAccountBaseC
...
+            BlobInventoryPolicyBlobSchemaField.CopyStatusDescription,
+            BlobInventoryPolicyBlobSchemaField.CustomerProvidedKeySha256,
+            BlobInventoryPolicyBlobSchemaField.RehydratePriority,
+            BlobInventoryPolicyBlobSchemaField.ArchiveStatus,
+            BlobInventoryPolicyBlobSchemaField.xmsblobsequencenumber,
+            BlobInventoryPolicyBlobSchemaField.EncryptionScope,
+            BlobInventoryPolicyBlobSchemaField.IncrementalCopy,
+            BlobInventoryPolicyBlobSchemaField.DeletionId,
+            BlobInventoryPolicyBlobSchemaField.DeletedTime,
+            BlobInventoryPolicyBlobSchemaField.TagCount,
```
Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/NewAzureStorageBlobInventoryPolicyRule.cs:125]: Please make sure the blob/container schemafields are aligned with the swagger description in https://github.com/Azure/azure-rest-api-specs/blob/894645e485d2f01e83999480d440e1bace3fec9b/specification/storage/resource-manager/Microsoft.Storage/stable/2021-09-01/storage.json#L5258

File: src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md [hunk 3]
```diff
@@ -16,13 +16,14 @@ Creates a blob inventory policy rule object, which can be used in Set-AzStorageB
 ```
 New-AzStorageBlobInventoryPolicyRule [-Name] <String> [-Disabled] -Destination <String> -Format <String>
  -Schedule <String> -BlobSchemaField <String[]> -BlobType <String[]> [-PrefixMatch <String[]>]
- [-IncludeSnapshot] [-IncludeBlobVersion] [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+ [-ExcludePrefix <String[]>] [-IncludeSnapshot] [-IncludeBlobVersion] [-IncludeDeleted]
```
Comment (blueww) [inline src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md:19]: Better to update example #1, #2 in https://github.com/Azure/azure-powershell/blob/ff4875a75c05c02bbc47f7cb13ea6e8937179ed6/src/Storage/Storage.Management/help/Set-AzStorageBlobInventoryPolicy.md

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageAccountTests.ps1 [hunk 4]
```diff
@@ -1770,67 +1770,69 @@ function Test-StorageBlobInventory
...
+        Assert-AreEqual 7 $policy1.Rules[2].Definition.SchemaFields.Count
+        Assert-Null $policy1.Rules[2].Definition.Filters.IncludeSnapshots
+        Assert-Null $policy1.Rules[2].Definition.Filters.IncludeBlobVersions
+        Assert-AreEqual $true $policy1.Rules[2].Definition.Filters.IncludeDeleted
+        Assert-AreEqual 2 $policy1.Rules[2].Definition.Filters.ExcludePrefix.Count
+        Assert-AreEqual 1 $policy1.Rules[2].Definition.Filters.BlobTypes.Count
+        Assert-AreEqual 2 $policy1.Rules[2].Definition.Filters.PrefixMatch.Count
+
+        # set policy with json and account name pipeline
+        $policy2 = Get-AzStorageAccount -ResourceGroupName $rgname  -StorageAccountName $stoname |  Set-AzStorageBlobInventoryPolicy -Policy (@{
```
Comment (blueww) [inline src/Storage/Storage.Management.Test/ScenarioTests/StorageAccountTests.ps1:1835]: Should we also update the test case for input from json

File: src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md]: Done

File: src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md (hunk 3 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/help/New-AzStorageBlobInventoryPolicyRule.md:19]: Done

File: src/Storage/Storage.Management/StorageAccount/NewAzureStorageBlobInventoryPolicyRule.cs (hunk 2 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/StorageAccount/NewAzureStorageBlobInventoryPolicyRule.cs:125]: Done.

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageAccountTests.ps1 (hunk 4 above)
Comment (yifanz0) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageAccountTests.ps1:1837]: Yes for sure. Updated the case.

**Review by blueww — APPROVED** [review]
LGTM

**Comment by Nickcandy** [comment]
It looks good to me. Please help double check @BethanyZhou

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**